url6f34cc586e_mekku93_sxl_interpreter_repl_tgz-pJ8-com_miguel_sxl_SXLJ8/src/com/miguel/sxl/SXLParser.java

```diff
@@ -0,0 +1,2739 @@
+  }
+
+/**== STATEMENTS ==*/
+  final public void IfStatement() throws ParseException {
+                      /*@bgen(jjtree) IfStatement */
+  ASTIfStatement jjtn000 = new ASTIfStatement(JJTIFSTATEMENT);
+  boolean jjtc000 = true;
+  jjtree.openNodeScope(jjtn000);
+  jjtn000.jjtSetFirstToken(getToken(1));
+    try {
+      jj_consume_token(KW_IF);
+      jj_consume_token(OPEN_PAREN);
+      Expression();
+      jj_consume_token(CLOSE_PAREN);
+                                                                  ASTBlock jjtn002 = new ASTBlock(JJTBLOCK);
+                                                                  boolean jjtc002 = true;
+                                                                  jjtree.openNodeScope(jjtn002);
+                                                                  jjtn002.jjtSetFirstToken(getToken(1));
+      try {
+                                                                    ASTThenStatement jjtn001 = new ASTThenStatement(JJTTHENSTATEMENT);
+                                                                    boolean jjtc001 = true;
+                                                                    jjtree.openNodeScope(jjtn001);
+                                                                    jjtn001.jjtSetFirstToken(getToken(1));
+        try {
+          Statement();
+        } catch (Throwable jjte001) {
+                                                                    if (jjtc001) {
+                                                                      jjtree.clearNodeScope(jjtn001);
+                                                                      jjtc001 = false;
+                                                                    } else {
+                                                                      jjtree.popNode();
+                                                                    }
+                                                                    if (jjte001 instanceof RuntimeException) {
+                                                                      {if (true) throw (RuntimeException)jjte001;}
+                                                                    }
+                                                                    if (jjte001 instanceof ParseException) {
+                                                                      {if (true) throw (ParseException)jjte001;}
+                                                                    }
+                                                                    {if (true) throw (Error)jjte001;}
+        } finally {
+                                                                    if (jjtc001) {
+                                                                      jjtree.closeNodeScope(jjtn001, true);
+                                                                      jjtn001.jjtSetLastToken(getToken(0));
+                                                                    }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 945:966, 1080:1101.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.